app/src/main/java/com/ncnf/views/fragments/group/GroupDisplayFragment.java

```diff
@@ -0,0 +1,157 @@
+        recycler.setLayoutManager(new LinearLayoutManager(getActivity()));
+
+        button = view.findViewById(R.id.open_map_button);
+
+        Bundle bundle = this.getArguments();
+        if(bundle != null) {
+            this.groupID = bundle.getString("GROUP_ID");
+
+            CompletableFuture<Group> thisGroup = repository.loadGroup(groupID);
+
+            thisGroup.thenAccept(group -> {
+                this.group = group;
+                prepareAllFields();
+            });
+
+        }
+    }
+
+    private void prepareAllFields() {
+
+        groupName.setText(group.getName());
+        groupDescription.setText(group.getDescription());
+        groupAddress.setText(group.getAddress());
+
+        button.setOnClickListener(v -> {
+            Intent intent = new Intent(getContext(), FriendsTrackerActivity.class);
+            intent.putExtra("GROUP_ID", group.getUuid().toString());
+            startActivity(intent);
+        });
+
+        userRepository.loadUser(group.getOwnerId()).thenAccept(user -> {
+            if(user.getFullName() != null && user.getFullName().length() > 0) {
+                if(user.getUsername() != null && user.getUsername().length() > 0) {
+                    groupOwner.setText(user.getFullName() + " (@" + user.getUsername() + ")");
+                }
+                else {
+                    groupOwner.setText(user.getFullName());
+                }
+            }
+            else {
+                if(user.getUsername() != null && user.getUsername().length() > 0) {
+                    groupOwner.setText("@" + user.getUsername());
+                }
+                else {
+                    groupOwner.setText(user.getEmail());
+                }
+            }
+        });
+
+        List<String> uuids = new ArrayList<>(group.getAttendees());
+        uuids.add(group.getOwnerId());
+        CompletableFuture<List<User>> users = userRepository.loadMultipleUsers(uuids);
+        users.thenAccept(users1 -> {
+            adapter = new UserListAdapter(getContext(), users1, this::showUserPublicProfile);
+            recycler.setAdapter(adapter);
+        });
+
+    }
+
+    private void showUserPublicProfile(User user) {
+
+        FragmentManager fragmentManager = getParentFragmentManager();
+
+        if(user.getUuid().equals(this.user.getUuid())) {
+            UserProfileTabFragment userProfileTabFragment = new UserProfileTabFragment();
+            fragmentManager.beginTransaction().replace(((ViewGroup) requireView().getParent()).getId(), userProfileTabFragment)
+                    .addToBackStack(null)
+                    .commit();
+        }
+        else {
+            PublicProfileFragment publicProfileFragment = new PublicProfileFragment(user);
+            fragmentManager.beginTransaction().replace(((ViewGroup) requireView().getParent()).getId(), publicProfileFragment)
+                    .addToBackStack(null)
+                    .commit();
+        }
+    }
+
+}
```

Comment:
I don't see any callback manager for having back key interaction. Shouldn't there be one to come back to the group view from the user profile/public profile view?

Reply:
From what I got from the internet, simply adding it to the stack suffices, and it works correctly when I run the app (i.e. goes back to group view) :)

Reply:
Oh ok, perfect then